### DISCUSSION TITLE
API endpoint params

### DISCUSSION THREAD

**Discussion by justindasilva** [nuxt/nuxt#18242] [Questions]
I'm loving a lot of things about `/server/api` and its integration into Nuxt.
However, I'm starting to note a few things missing which I think should be standard.

1. Do we have some way to grab parameters in the API URL? For example, grabbing the `id` parameter from a URL that looks like `/api/page/view/:id`?
Was hoping that in the code, I can do something like:
```
import { useParams } from 'h3'

export default async (req, res) => {
    
    const params = useParams(req)
    const id = params.id
}
```
Perhaps the filename in the codebase will have a path like: `/server/api/page/view/_id.ts` which kinda looks similar to what we are able to do in the `pages` directory.

2. Also, I think it would be nice to support camelCase naming of endpoints. Something like `/api/page/updateById/:id`

3. Also, unless I'm missing something, I'm not sure how to control GET/POST/PUT/etc requests.

**Comment by wobsoriano** [comment]
Check this https://github.com/nuxt/framework/discussions/1137

This isn't implemented by [h3](https://github.com/unjs/h3) yet
